File: ios-app/CalendarioCiclismo/Models/ChampionshipCountry.swift
import Foundation

/// Un país en la rejilla del Modo Campeonatos: bandera + sede + sus pruebas
/// emparejadas con su jornada (`EnrichedRaceDay`) por slot.
/// Equivalente a `byCountry[cc]` en `js/campeonatos.js`.
struct ChampionshipCountry: Identifiable, Hashable {
    let countryCode: String
    let hostCity: String?
    /// Pruebas presentes para este país, indexadas por slot.
    let slots: [ChampionshipsConfig.Slot: EnrichedRaceDay]

    var id: String { countryCode }

    /// Slots visibles bajo un filtro, en el orden fijo de columna. El filtro
    /// `today` no restringe por slot sino por fecha (la jornada del día en curso).
    func visibleSlots(filter: ChampionshipsConfig.Filter) -> [ChampionshipsConfig.Slot] {
        if filter == .today {
            let today = DateFormatting.todayKey()
            return ChampionshipsConfig.Slot.allCases.filter { slots[$0]?.raceDay.dateKey == today }
        }
        let allowed = Set(filter.slots)
        return ChampionshipsConfig.Slot.allCases.filter { allowed.contains($0) && slots[$0] != nil }
    }
}

File: android-app/app/src/main/java/app/calendariociclismo/android/data/model/ChampionshipCountry.kt
package app.calendariociclismo.android.data.model

import app.calendariociclismo.android.util.ChampionshipsConfig
import app.calendariociclismo.android.util.DateFormatting

/**
 * Un país en la rejilla del Modo Campeonatos: bandera + sede + sus pruebas
 * emparejadas con su jornada (`EnrichedRaceDay`) por slot.
 * Equivalente a `byCountry[cc]` en `js/campeonatos.js`.
 */
data class ChampionshipCountry(
    val countryCode: String,
    val hostCity: String?,
    val slots: Map<ChampionshipsConfig.Slot, EnrichedRaceDay>,
) {
    /** Slots visibles bajo un filtro, en el orden fijo de columna. El filtro
     *  `TODAY` no restringe por slot sino por fecha (la jornada del día en curso). */
    fun visibleSlots(filter: ChampionshipsConfig.Filter): List<ChampionshipsConfig.Slot> {
        if (filter == ChampionshipsConfig.Filter.TODAY) {
            val today = DateFormatting.todayKey()
            return ChampionshipsConfig.Slot.values().filter { slots[it]?.raceDay?.dateKey == today }
        }
        val allowed = filter.slots.toSet()
        return ChampionshipsConfig.Slot.values().filter { it in allowed && slots[it] != null }
    }
}
